### PR TITLE
kubernetes: fixing credentials propagation for cluster suppliers

### LINKED ISSUE #23876: 🐛 Bug Report: Error 500 Kubernetes Plugin
### 📜 Description

The backend returns a 500 when listing the Kubernetes resources associated with a Component.

### 👍 Expected behavior

List the Kubernetes resources associated with a Component.

### 👎 Actual Behavior with Screenshots

image

### 👟 Reproduction steps

1. Go on a Component
2. Go to the Kubernetes Tab

### 📃 Provide the context for the Bug.

When opening the Kubernetes tab, the frontend successfully receives the list of clusters on the `api/kubernetes/clusters` endpoint. The 500 is returned when the resources are looked up on the service endpoint `api/kubernetes/services/hello-world`

The frontend shows:
```
Errors: Request failed with 500 , {"error":"Cannot read properties of undefined (reading 'credentials')"}
```

On the backend, we can see a bit more details:
```
"level":"error","message":"action=retrieveObjectsByServiceId service=hello-world, error=TypeError: Cannot read properties of undefined (reading 'credentials')","plugin":"kubernetes","service":"backstage"}
```

With 1.24 using backend authentification by default, it appears the cluster-locator was [updated](https://github.com/backstage/backstage/commit/69d0217cba0e00bdf59194872ac91f1767273e17) to allow passing the `option.credentials` parameters on `clusterSupplier.getClusters`; I guess that the service-locator should have something similar as it currently uses `clusterSupplier.getClusters` without passing the Backstage Credential which cause the error seen in the backend logs (options is undefined):
- [MultiTenantServiceLocator](https://github.com/backstage/backstage/blob/560c6854cbe01a9f19daee5450fdce4fddba30ef/plugins/kubernetes-backend/src/service-locator/MultiTenantServiceLocator.ts#L39)
- [SingleTenantServiceLocator](https://github.com/backstage/backstage/blob/560c6854cbe01a9f19daee5450fdce4fddba30ef/plugins/kubernetes-backend/src/service-locator/SingleTenantServiceLocator.ts#L39-L52)
- [CatalogRelationServiceLocator](https://github.com/backstage/backstage/blob/560c6854cbe01a9f19daee5450fdce4fddba30ef/plugins/kubernetes-backend/src/service-locator/CatalogRelationServiceLocator.ts#L44-L51)

### 🖥️ Your Environment

OS:   Darwin 23.4.0 - darwin/arm64
node: v18.16.0
yarn: 1.22.21
cli:  0.26.1 (installed)
backstage:  1.24.0

Dependencies:
  @backstage/app-defaults                                          1.5.2
  @backstage/backend-app-api                                       0.5.14, 0.6.1
  @backstage/backend-common                                        0.20.2, 0.21.5
  @backstage/backend-defaults                                      0.2.15
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.8
  @backstage/backend-plugin-api                                    0.6.15
  @backstage/backend-tasks                                         0.5.20
  @backstage/backend-test-utils                                    0.3.5
  @backstage/catalog-client                                        1.6.2
  @backstage/catalog-model                                         1.4.5
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.4
  @backstage/cli                                                   0.25.2, 0.26.1
  @backstage/config-loader                                         1.7.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.2
  @backstage/core-compat-api                                       0.2.2
  @backstage/core-components                                       0.14.2
  @backstage/core-plugin-api                                       1.9.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.6
  @backstage/frontend-plugin-api                                   0.6.2
  @backstage/integration-aws-node                                  0.1.11
  @backstage/integration-react                                     1.1.25
  @backstage/integration                                           1.9.1
  @backstage/plugin-api-docs                                       0.11.2
  @backstage/plugin-app-backend                                    0.3.63
  @backstage/plugin-app-node                                       0.1.15
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.7
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.7
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.10
  @backstage/plugin-auth-backend-module-github-provider            0.1.12
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.12
  @backstage/plugin-auth-backend-module-google-provider            0.1.12
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.10
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.12
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.8
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.6
  @backstage/plugin-auth-backend-module-okta-provider              0.0.8
  @backstage/plugin-auth-backend                                   0.22.2
  @backstage/plugin-auth-node                                      0.4.10
  @backstage/plugin-auth-react                                     0.0.2
  @backstage/plugin-catalog-backend-module-github                  0.5.6
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.13
  @backstage/plugin-catalog-backend                                1.20.0
  @backstage/plugin-catalog-common                                 1.0.22
  @backstage/plugin-catalog-graph                                  0.4.2
  @backstage/plugin-catalog-import                                 0.10.8
  @backstage/plugin-catalog-node                                   1.10.0
  @backstage/plugin-catalog-react                                  1.11.1
  @backstage/plugin-catalog                                        1.18.1
  @backstage/plugin-events-node                                    0.3.1
  @backstage/plugin-github-actions                                 0.6.13
  @backstage/plugin-home-react                                     0.1.10
  @backstage/plugin-home                                           0.6.2
  @backstage/plugin-kubernetes-common                              0.7.5
  @backstage/plugin-kubernetes-node                                0.1.9
  @backstage/plugin-kubernetes-react                               0.3.2
  @backstage/plugin-kubernetes                                     0.11.7
  @backstage/plugin-org                                            0.6.22
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.12
  @backstage/plugin-permission-backend                             0.5.39
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.26
  @backstage/plugin-permission-react                               0.4.21
  @backstage/plugin-proxy-backend                                  0.4.13
  @backstage/plugin-scaffolder-backend-module-azure                0.1.7
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.5
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.5
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.5
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.7
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.5
  @backstage/plugin-scaffolder-backend-module-github               0.2.5
  @backstage/plugin-scaffolder-backend-module-gitlab               0.3.1
  @backstage/plugin-scaffolder-backend                             1.22.2
  @backstage/plugin-scaffolder-common                              1.5.1
  @backstage/plugin-scaffolder-node                                0.4.1
  @backstage/plugin-scaffolder-react                               1.8.2
  @backstage/plugin-scaffolder                                     1.19.1
  @backstage/plugin-search-backend-module-catalog                  0.1.20
  @backstage/plugin-search-backend-module-pg                       0.5.24
  @backstage/plugin-search-backend-module-techdocs                 0.1.20
  @backstage/plugin-search-backend-node                            1.2.19
  @backstage/plugin-search-backend                                 1.5.5
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.8
  @backstage/plugin-search                                         1.4.8
  @backstage/plugin-tech-radar                                     0.7.1
  @backstage/plugin-techdocs-backend                               1.10.2
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.7
  @backstage/plugin-techdocs-node                                  1.12.1
  @backstage/plugin-techdocs-react                                 1.2.1
  @backstage/plugin-techdocs                                       1.10.2
  @backstage/plugin-user-settings                                  0.8.3
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.5.2
  @backstage/theme                                                 0.5.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by benjdlambert**
Fixes #23876

### CODE REVIEW

File: plugins/kubernetes-backend/src/service/KubernetesProxy.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ export type KubernetesProxyOptions = {
   logger: Logger;
   clusterSupplier: KubernetesClustersSupplier;
   authStrategy: AuthenticationStrategy;
+  httpAuth: HttpAuthService;
```
Comment (Rugvip) [inline plugins/kubernetes-backend/src/service/KubernetesProxy.ts:84]: Ugh, this needs to be optional with a fallback to `createLegacyAuthAdapters` which then forces us to depend on the `DiscoveryService`. Callers of the constructor are coming from the old backend system and won't have a `HttpAuthService` instance.

**Review by Rugvip — CHANGES_REQUESTED** [review]
👍, just some trouble with the `KubernetesProxy` to figure out

**Comment by mclarke47** [comment]
Thanks for the fix, feel free to merge or I can help review next week

**Comment by benjdlambert** [comment]
@mclarke47 no worries buddy! Here to help :tada:

**Review by Rugvip — APPROVED** [review]
👍 , :shipit:

**Comment by clementblaise** [comment]
Thanks for providing an update so rapidly @benjdlambert working as expected

**Comment by swnia** [comment]
Would it be possible to add this change to 1.25.X(3) as it fixes a now broken part? 1.26.0 is over a month away...

**Comment by benjdlambert** [comment]
@swnia the next mainline release is actually 16th of April, our tooling broke a little bit when we did a bonus release. https://github.com/backstage/backstage/pull/24015/files

This will also be released part of the `-next` release train going out later today.

**Comment by swnia** [comment]
That's great to hear! Thanks a lot

**Pull request merged**